libs/datamodel/core/src/transform/ast_to_dml/db.rs

```diff
@@ -54,7 +54,7 @@ pub(crate) struct ParserDatabase<'ast> {
     ast: &'ast ast::SchemaAst,
     datasource: Option<&'ast Datasource>,
     names: Names<'ast>,
-    types: Types<'ast>,
+    pub(crate) types: Types<'ast>,
```

Comment:
Just starting the read-through, so it may become apparent why later, but I think we should really strive to keep this private.